internal/pkg/agent/install/uninstall_windows.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package install
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+func isBlockingOnExe(err error) bool {
+	if err == nil {
+		return false
+	}
+	path := getPathFromError(err)
+	return path != ""
+}
+
+func removeBlockingExe(blockingErr error) (string, error) {
+	path := getPathFromError(blockingErr)
+	if path == "" {
+		return "", nil
+	}
+
+	// open handle for delete only
+	h, err := openDeleteHandle(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open handler for %q: %w", path, err)
+	}
+
+	// rename handle
+	err = renameHandle(h)
+	if err != nil {
+		_ = windows.CloseHandle(h)
+		return "", fmt.Errorf("failed to rename handler for %q: %w", path, err)
+	}
+	_ = windows.CloseHandle(h)
+
+	// re-open handle
+	h, err = openDeleteHandle(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open handler after rename for %q: %w", path, err)
+	}
+
+	// disposite the handle
+	err = depositeHandle(h)
+	if err != nil {
+		_ = windows.CloseHandle(h)
+		return "", fmt.Errorf("failed to deposite handler for %q: %w", path, err)
+	}
+	_ = windows.CloseHandle(h)
```

Comment:
```suggestion
	err = disposeHandle(h)
	_ = windows.CloseHandle(h)
	if err != nil {
		return "", fmt.Errorf("failed to dispose handler for %q: %w", path, err)
	}
```

Should be `dispose` in the name going off of the `FileDispositionInfo` later on. Also looks like we can unconditionally close the handle before checking the error.